keystore/src/connection/platform/wasm/migration.rs

```diff
@@ -193,6 +273,99 @@ where
     }
 }
 
+trait WasmMigrationUniqueExt: UniqueEntity {
+    type EntityTypeV1_0_0: UniqueEntityV1_0_0<ConnectionType = KeystoreDatabaseConnectionV1_0_0>;
+    async fn convert_from_v_1_2_0_or_earlier(
+        connection: &mut KeystoreDatabaseConnectionV1_0_0,
+    ) -> CryptoKeystoreResult<Vec<Self>> {
+        let old_record_result = Self::EntityTypeV1_0_0::find_unique(connection).await;
+        match old_record_result {
+            Ok(old_record) => {
+                let serialized = postcard::to_stdvec(&old_record)?;
+                let new_record = postcard::from_bytes::<Self>(&serialized)?;
+                Ok(vec![new_record])
+            }
+            // When it doesn't exist, it doesn't need conversion.
+            Err(CryptoKeystoreErrorV1_0_0::NotFound(..)) => Ok(vec![]),
+            Err(e) => Err(e)?,
+        }
+    }
+}
+
+impl WasmMigrationExt for E2eiEnrollment {
+    type EntityTypeV1_0_0 = E2eiEnrollmentV1_0_0;
+    /// This entity doesn't need migration. It's fine to start a new enrollment after migration.
+    async fn convert_from_v_1_2_0_or_earlier(
+        _connection: &mut KeystoreDatabaseConnectionV1_0_0,
+    ) -> CryptoKeystoreResult<Vec<Self>> {
+        Ok(vec![])
+    }
+}
+
+impl WasmMigrationUniqueExt for E2eiRefreshToken {
+    type EntityTypeV1_0_0 = E2eiRefreshTokenV1_0_0;
+}
+
+impl WasmMigrationUniqueExt for E2eiAcmeCA {
+    type EntityTypeV1_0_0 = E2eiAcmeCAV1_0_0;
+}
+
 impl WasmMigrationExt for MlsCredential {
     type EntityTypeV1_0_0 = MlsCredentialV1_0_0;
 }
+
+impl WasmMigrationExt for MlsSignatureKeyPair {
+    type EntityTypeV1_0_0 = MlsSignatureKeyPairV1_0_0;
+}
+
+impl WasmMigrationExt for MlsHpkePrivateKey {
+    type EntityTypeV1_0_0 = MlsHpkePrivateKeyV1_0_0;
+}
+
+impl WasmMigrationExt for MlsEncryptionKeyPair {
+    type EntityTypeV1_0_0 = MlsEncryptionKeyPairV1_0_0;
+}
+
+impl WasmMigrationExt for MlsEpochEncryptionKeyPair {
+    type EntityTypeV1_0_0 = MlsEpochEncryptionKeyPairV1_0_0;
+}
+
+impl WasmMigrationExt for MlsPskBundle {
+    type EntityTypeV1_0_0 = MlsPskBundleV1_0_0;
+}
+
+impl WasmMigrationExt for MlsKeyPackage {
+    type EntityTypeV1_0_0 = MlsKeyPackageV1_0_0;
+}
+
+impl WasmMigrationExt for PersistedMlsGroup {
+    type EntityTypeV1_0_0 = PersistedMlsGroupV1_0_0;
+}
+
+impl WasmMigrationExt for PersistedMlsPendingGroup {
+    type EntityTypeV1_0_0 = PersistedMlsPendingGroupV1_0_0;
+}
```

Comment:
I don't understand why we have a bunch of impls of `WasmMigrationExt` for entity types that don't have `find_unique`. Could you please explain how this works?